python-sdk/tests/sql/operators/test_cleanup.py

```diff
@@ -186,16 +186,26 @@ def test_cleanup_multiple_table(database_table_fixture, multiple_tables_fixture)
     ids=["two_tables"],
 )
 def test_cleanup_default_all_tables(sample_dag, database_table_fixture, multiple_tables_fixture):
-    @aql.transform
+    @aql.transform()
     def foo(input_table: Table):
         return "SELECT * FROM {{input_table}}"
 
+    db, _ = database_table_fixture
     table_1, table_2 = multiple_tables_fixture
+    assert db.table_exists(table_1)
+    assert db.table_exists(table_2)
+
+    def assert_tables_do_not_exist():
+        assert not db.table_exists(table_1)
+        assert not db.table_exists(table_2)
+
     with sample_dag:
         foo(table_1)
         foo(table_2)
 
-        aql.cleanup([])
+        # FIXME: on_success_callback is being ignored
+        # see https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/callbacks.html#callbacks
+        aql.cleanup(on_success_callback=assert_tables_do_not_exist)
```

Comment:
@dimberman any idea how we can test this? I want to assert the tables have been removed after running `aql.cleanup`. The solution I would have loved to work is by using `on_success_callback`. But unfortunately callbacks are being ignored by our (& Airflow's) runner.

Reply:
Probably one solution until `on_*_callback` is supported might be a follow-up python task that gets the table names from cleanup (as I think `aql.cleanup` pushes table names to XCom -- do verify it though) and checks if they exists?

Reply:
Good point, but I already tested similar ways, but none of them worked:
* create a task downstream to run `assert` -> this does not work as cleanup waits until the dagrun is completed
* create another dag that waits until the dag finishes and then runs `assert` -> this did not work as we delete the dag after running it, it seems (but not sure)

Reply:
How about a non-Airflow command after `run_dag` to assert this manually by calling the DB?

Reply:
So you mean kind of a sensoring operation but not as an airflow task?

Reply:
No I just meant the following on L210:

```
assert not db.table_exists(table_1)
assert not db.table_exists(table_2)
```

Reply:
That was the first thing I tried 😂 This does not work as the cleanup waits until the dagrun has finished. These asserts would check before cleanup would have a chance to remove the tables.

Reply:
But on L210, the DAG will have completed the execution. It will only be running in case of a "failed test" to prevent which you can add a "timeout" of few seconds.

Reply:
@feluelle do you want to pair on this tomorrow?

Reply:
Yes, let's pair on this :)

Reply:
Thanks @kaxil, @dimberman has found the solution. I was testing for the wrong tables to not exist. I have pushed the changes. All tests are passing.